### PR TITLE
Use new matches filter for global search + search-sheet + card chooser

### PR DESCRIPTION
**Pull request by lukemelia**
## Summary

- Swap host global search / search-sheet query assembly from `{ contains: { cardTitle } }` to `{ matches }` so content (not just titles) drives results.
- Single change site: `buildSearchQuery` in `packages/host/app/utils/card-search/query-builder.ts` — existing `cardType`/realm scoping clauses preserved.
- Add unit tests for `buildSearchQuery` covering empty/whitespace keys, plain search, `baseFilter` mode, and type-filter composition.

Stacked on #4458 (CS-10825 — `matches` filter plumbing).

Linear: [CS-10833](https://linear.app/cardstack/issue/CS-10833/host-swap-global-search-search-sheet-from-title-like-to-matches)

## Out of scope

- Card chooser dialogs (separate issue)
- Snippet rendering in result rows (separate issue)

## Test plan

- [ ] New unit tests pass: `Unit | card-search/query-builder`
- [ ] Acceptance tests that drive the search-sheet (e.g. `interact-submode-test.gts`) still find cards by title — they use single-word titles that FTS will tokenize, so ordering stays predictable.
- [ ] Manual: type a word that appears only in a card's body (not its title) — the card surfaces in search results.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/host/tests/integration/components/card-catalog-test.gts [hunk 1]
```diff
@@ -434,4 +437,36 @@ module('Integration | card-catalog', function (hooks) {
       assert.dom('[data-test-stack-card-index="1"]').doesNotExist();
     });
   });
+
+  module('content search', function () {
+    test(`finds cards by description content, not just title, and respects type scoping`, async function (assert) {
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/host/tests/integration/components/card-catalog-test.gts]: Applied the suggestion in 5238b9b.

**Review by habdelra — APPROVED** [review]

**Review by FadhlanR — APPROVED** [review]
